llvm/include/llvm/ADT/STLExtras.h

```diff
@@ -1261,6 +1261,10 @@ template <typename ContainerTy> auto make_second_range(ContainerTy &&c) {
       });
 }
 
+template <typename R> auto make_address_range(R &&Range) {
```

Comment:
Not sure we want to have `Range` to be a universal reference. If it's a temporary I think that its lifetime will not be extended and we'll end up building a range of addresses that point to objects whose liftime has ended.

Reply:
You mean like `make_address_range(std::vector { 1, 2, 3 })`?

AFAIU that's a mistake on the caller's side and there's not much we can do about it, except returning for copy the vector, but ranges are just pair of iterators, they are not meant to be owning.

Reply:
My point is that if you change the prototype to (notice the single `&`, it's an lvalue-reference here, not a universal reference anymore)
```c++
template <typename R> auto make_address_range(R &Range)
```
then the compiler forbids you to pass an rvalue reference to `make_address_range`, turning runtime errors into compile time errors. See [here](https://godbolt.org/z/a5vdqbWeh) for a simplified example.

Reply:
Anyway, if we do like I propose we lose ability to use this with non-owning ranges, so let's leave it like you propose. Also, this problem will be solved in C++23 AFAIU.